### ISSUE TITLE
Crash when meta: { client: { env: { server: false } } }

### ISSUE THREAD

**Issue by jonaskuske** [vikejs/vike#1693]
### Description

My app crashes on startup when I add:

```ts
// renderer/+config.ts
import type { Config } from 'vike/types'

export default {
  meta: {
    client: { env: { server: false } },
  },
} satisfies Config
```

However, if I first start the app and then adjust the client meta config, the app automatically reloads and correctly loads or ceases to load `+client.ts` files depending on the configured value.

Error:

```
Failed to start server: Error: [vike@0.4.175][Bug] You stumbled upon a Vike bug. Go to https://github.com/vikejs/vike/issues/new and copy-paste this error. A maintainer will fix the bug (usually under 24 hours).
    at getConfigValue (file:///home/jonas/github.com/jonaskuske/thesis/.yarn/__virtual__/vike-virtual-5a5b678e33/0/cache/vike-npm-0.4.175-e0b0805341-691a10e37b.zip/node_modules/vike/dist/esm/shared/page-configs/getConfigValueBuildTime.js:25:5)
    at getConfigValueBuildTime (file:///home/jonas/github.com/jonaskuske/thesis/.yarn/__virtual__/vike-virtual-5a5b678e33/0/cache/vike-npm-0.4.175-e0b0805341-691a10e37b.zip/node_modules/vike/dist/esm/shared/page-configs/getConfigValueBuildTime.js:7:25)
    at file:///home/jonas/github.com/jonaskuske/thesis/.yarn/__virtual__/vike-virtual-5a5b678e33/0/cache/vike-npm-0.4.175-e0b0805341-691a10e37b.zip/node_modules/vike/dist/esm/node/plugin/plugins/buildConfig.js:167:33
    at Array.forEach (<anonymous>)
    at analyzeClientEntries (file:///home/jonas/github.com/jonaskuske/thesis/.yarn/__virtual__/vike-virtual-5a5b678e33/0/cache/vike-npm-0.4.175-e0b0805341-691a10e37b.zip/node_modules/vike/dist/esm/node/plugin/plugins/buildConfig.js:153:17)
    at getVirtualFiles (file:///home/jonas/github.com/jonaskuske/thesis/.yarn/__virtual__/vike-virtual-5a5b678e33/0/cache/vike-npm-0.4.175-e0b0805341-691a10e37b.zip/node_modules/vike/dist/esm/node/plugin/plugins/devConfig/determineOptimizeDeps.js:83:67)
    at determineOptimizeDeps (file:///home/jonas/github.com/jonaskuske/thesis/.yarn/__virtual__/vike-virtual-5a5b678e33/0/cache/vike-npm-0.4.175-e0b0805341-691a10e37b.zip/node_modules/vike/dist/esm/node/plugin/plugins/devConfig/determineOptimizeDeps.js:17:37)
    at configResolved (file:///home/jonas/github.com/jonaskuske/thesis/.yarn/__virtual__/vike-virtual-5a5b678e33/0/cache/vike-npm-0.4.175-e0b0805341-691a10e37b.zip/node_modules/vike/dist/esm/node/plugin/plugins/devConfig/index.js:57:17)
```

**Comment by brillout** [comment]
What is it you're trying to achieve with `meta.client.env = { server: false, client: false }`?

I will improve the error message.

**Comment by jonaskuske** [comment]
This: https://github.com/vikejs/vike/discussions/1686#discussioncomment-9737899 – only load `+client.ts` scripts when the app is run in MPA mode. And as described, it works perfectly when I change the config during runtime. But if it's there on startup, it chokes.

**Comment by brillout** [comment]
Can you do this instead?

```js
// +config.js

import client from './client.jsx'

export default {
  client: process.env.SOME_ENV ? null : client
}
```

The `.jsx` file extension is needed, see https://vike.dev/config#pointer-imports and https://github.com/vikejs/vike/issues/1500.

**Comment by brillout** [comment]
If you remove the assertion at `file:///home/jonas/github.com/jonaskuske/thesis/.yarn/__virtual__/vike-virtual-5a5b678e33/0/cache/vike-npm-0.4.175-e0b0805341-691a10e37b.zip/node_modules/vike/dist/esm/shared/page-configs/getConfigValueBuildTime.js:25:5` does everything work then?

**Comment by jonaskuske** [comment]
I'll give it a try later, thanks!

Otherwise the manual pointer import should be okay for now as I don't have too many `+client.ts` files, but not a perfect solution as I can't configure in one place but have to add a separate `+config.ts` for every `+client.ts` file.

**Comment by jonaskuske** [comment]
Yup, everything works once I remove that assertion!

(https://github.com/jonaskuske/thesis/commit/1b4af53da38d3fe56fedd75dd2790e30f562a861 and https://github.com/jonaskuske/thesis/commit/8fe069c4bf9f89883532dcb9f8c47b18ca0bcffb)

**Comment by brillout** [comment]
Fix pre-released as `0.4.176-commit-775d83b`.
